Fix result check on readAnnotations(...)

fengxue-IS opened the pull request:
check if the return value is BCT_ERR_OUT_OF_ROM and return the
error to increase buffer size rather than marking annotation as
malformed.

Close: #1588 

Signed-off-by: Jack Lu <Jack.S.Lu@ibm.com>

DanHeidinga commented on runtime/bcutil/cfreader.c:
```diff
@@ -528,7 +530,9 @@ readAttributes(J9CfrClassFile * classfile, J9CfrAttribute *** pAttributes, U_32
-				if (result != 0) {
+				if (BCT_ERR_OUT_OF_ROM == result) {
+					return result;
+				} else if (result != 0) {
```
Can you change this to `(BCT_ERR_NO_ERROR != result)`?  Using the named error code is clearer and allows changing the value in the future

DanHeidinga replied:
Is this missing a `index != end` check?

fengxue-IS replied:
This check is inside a loop, so `index != end` isn't required as it will break out and be checked in more detail, see line 543 below.
I will update the 0 to `BCT_ERR_NO_ERROR`, thanks for pointing out

DanHeidinga replied:
Agreed.

DanHeidinga commented:
Does the test in #1588 pass with this change?

fengxue-IS commented:
The test case mentioned in #1588 passes with this change added.

fengxue-IS commented:
updated the commit with comments, moved the check outside of the loop to be more clear/consistent, also added this check for `readTypeAnnotation(...)` as this also calls `ALLOC_CAST` and `readAnnotations(...)`

DanHeidinga commented:
Jenkins test sanity plinux jdk8
